bemani/backend/iidx/pendual.py

```diff
@@ -1707,6 +1741,172 @@ def format_profile(self, userid: UserID, profile: Profile) -> Node:
+        time_sandglass.set_attribute('item_num', "99")
+        #TODO: Make these purchasable from the WEBUI using paseli. Game will send ``timesandglass_consume`` to subtract from account
+
+        present_time_sandglass = Node.void('present_time_sandglass')
+        root.add_child(present_time_sandglass)
+        present_time_sandglass.set_attribute('item_num', "99")
+        #TODO: Same as above.
+
+        pointsdict = profile.get_dict('point_events') #used for summer, dcat, and pyramid
+        #destiny event
+        dcat = Node.void('destiny_catharsis')
+        root.add_child(dcat)
+        dcat.set_attribute('music_bit', str(pointsdict.get_int('dcat_pts', 0)))
+
+        #BEMANI SUMMER 2015 Event
+        bsum = Node.void('bemani_summer_collabo')
+        root.add_child(bsum)
+        bsum.set_attribute('music_bit', str(pointsdict.get_int('summer_pts', 0)))
+
+        #Chrono Diver event (yaaaay!)
+        diverdict = profile.get_dict('diver_dict')
+        cdive = Node.void('chrono_diver')
+        root.add_child(cdive)
+        cdive.set_attribute('play_count', str(diverdict.get_int('play_count', 0)))
+        cdive.set_attribute('present_unlock', str(diverdict.get_int('present_unlock', 0)))
+        cdive.set_attribute('future_unlock', str(diverdict.get_int('future_unlock', 0)))
+        cdive.set_attribute('success_count_0_n', str(diverdict.get_int('success_count_0_n', 0)))
+        cdive.set_attribute('success_count_0_h', str(diverdict.get_int('success_count_0_h', 0)))
+        cdive.set_attribute('success_count_0_a', str(diverdict.get_int('success_count_0_a', 0)))
+        cdive.set_attribute('success_count_1_n', str(diverdict.get_int('success_count_1_n', 0)))
+        cdive.set_attribute('success_count_1_h', str(diverdict.get_int('success_count_1_h', 0)))
+        cdive.set_attribute('success_count_1_a', str(diverdict.get_int('success_count_1_a', 0)))
+        cdive.set_attribute('success_count_2_n', str(diverdict.get_int('success_count_2_n', 0)))
+        cdive.set_attribute('success_count_2_h', str(diverdict.get_int('success_count_2_h', 0)))
+        cdive.set_attribute('success_count_2_a', str(diverdict.get_int('success_count_2_a', 0)))
+        cdive.set_attribute('success_count_3_n', str(diverdict.get_int('success_count_3_n', 0)))
+        cdive.set_attribute('success_count_3_h', str(diverdict.get_int('success_count_3_h', 0)))
+        cdive.set_attribute('success_count_3_a', str(diverdict.get_int('success_count_3_a', 0)))
+        cdive.set_attribute('story_list', str(diverdict.get_int('story_list', 0)))
+
+        #konami pyramid scheme
+        pyramid = Node.void('pyramid')
+        root.add_child(pyramid)
+        pyramid.set_attribute('music_list', "1")
+        pyramid.set_attribute('item_list', "1")
+        pyramid.set_attribute('statue_0', str(0))
+        pyramid.set_attribute('statue_1', str(0))
+        pyramid.set_attribute('statue_2', str(0))
+        #TODO: Figure out how to actually send this data. 
```

Comment:
I gave up reviewing here. This seems like a bunch of hardcoded stuff to make the event *show up* but it doesn't actually implement any of the logic to make the event *work*.

Reply:
This does not cause the pyramid event to show up. i suppose i should have removed them from my PR as they're basically useless as they stand and are not required. From what i can tell, this event will only trigger on an older build of pendual, so i might as well just not bother with it at all.